bigquery/user_defined_functions/string-functions.sql

```diff
@@ -0,0 +1,109 @@
+# Contains useful string processing functions that can be reused across multiple scheduled queries and views in BigQuery.
+# This script must be run in BigQuery for these functions to become available, and re-run to overwrite the available functions with any new version.
+  
+# Take a client user agent from a Cloudfront server log and use it to categorise that traffic into a device category
+CREATE OR REPLACE FUNCTION
+  `teacher-vacancy-service.production_dataset.convert_client_user_agent_to_device_category`(client_user_agent STRING)
+  RETURNS STRING AS (
+    CASE
+      WHEN LOWER(client_user_agent) LIKE "%bot%" OR LOWER(client_user_agent) LIKE "%http%" OR LOWER(client_user_agent) LIKE "%python%" OR LOWER(client_user_agent) LIKE "%scan%" OR LOWER(client_user_agent) LIKE "%check%" OR LOWER(client_user_agent) LIKE "%spider%" OR LOWER(client_user_agent) LIKE "%curl%" OR LOWER(client_user_agent) LIKE "%trend%" OR LOWER(client_user_agent) LIKE "%fetch%" THEN "bot"
```

Comment:
Could you split the bot line out onto separate lines like the mobile line?

Reply:
Helps my reading brain make an analogy between the two if they look similar

Reply:
Done!